actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -94,13 +94,17 @@ def authenticate(request, &login_procedure)
       end
 
       def has_basic_credentials?(request)
-        request.authorization.present? && (auth_scheme(request).downcase == 'basic')
+        request.authorization.present? && (auth_scheme(request).downcase == 'basic') && !user_name_or_password_blank?(request)
       end
 
       def user_name_and_password(request)
         decode_credentials(request).split(':', 2)
       end
 
+      def user_name_or_password_blank?(request)
```

Comment:
By the way `all?` will return false if any of the elements returns false(http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) so actually the name matches the implementation as far as I see.